core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -359,4 +516,58 @@ class SpendReportingService(
         case ex: Exception =>
           Future.failed(RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.InternalServerError, ex)))
       }
+
+  def getSpendForAllWorkspaces(
+    start: DateTime,
+    end: DateTime
+  ): Future[List[SpendReportingResults]] = {
+    validateReportParameters(start, end)
+    for {
+      workspaces <- getOwnerWorkspaces()
+      projectNames = workspaces
+        .map(wsResp =>
+          wsResp.workspace.googleProject -> WorkspaceName(wsResp.workspace.namespace, wsResp.workspace.name)
+        )
+        .toMap // Map[GoogleProjectId, WorkspaceName]
```

Comment:
For others watching the PR: this is where Bria and I discussed changing the business logic. Is it feasible to:
1. Query Sam's `listResourcesV2` for all `billing-project` resources on which the user has the `read_spend_report` action
2. Query the Rawls db to retrieve the workspace names/ids associated with each of those billing projects
3. (maybe) Query Sam's `listResourcesV2` for all `workspace` resources on which the user has the `read` action, to verify that the user should be able to see all the workspace names from step 2

Is there ever a case in which a user will have `read_spend_report` on a `billing-project` but cannot read the workspace associated with that project?